### PR TITLE
Implemented basic procedure builder

### LINKED ISSUE #112: Edit page interface
Each page should have an element by default.

For questions with choices:
- Answer field should be a dropdown of list of available choices
- Choices should an editable/sortable list of choices (draggable to reorder, press enter to create new item)

For questions without choices
- Answer should be hidden for non-relevant questions (e.g. GPS)
- Choices should be hidden

### PR DESCRIPTION
**Pull request by Trinovantes**
This builder is now on par with our old Ember client plus a lot of
features/enhancements based on feedback from a few months ago.

Closes #112, #170, #197, #216

### CODE REVIEW

**Comment by Trinovantes** [comment]
Initial state:
screenshot from 2015-12-11 19 54 51

Create new page:
screenshot from 2015-12-11 19 55 37

Create new element:
screenshot from 2015-12-11 19 56 16

Another view:
screenshot from 2015-12-11 19 57 08

File: src-backbone/app/js/behaviors/sortable_collection_behavior.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            .disableSelection()
+        ;
+    },
+
+    onAddChild: function(view) {
+        view.$el.attr('data-model-cid', view.model.cid);
+    },
+
+    _get_child_view_container: function() {
+        if (typeof this.view.getChildViewContainer == 'function') {
```
Comment (ConnorCimowsky) [inline src-backbone/app/js/behaviors/sortable_collection_behavior.js]: do we prefer `===`?

File: src-backbone/app/js/models/page_model.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    is_active: function() {
+        let active_page_id = this.collection.parent_procedure.active_page_id;
+        return _.isNumber(active_page_id) && this.get('id') === active_page_id;
+    },
+
+    create_new_element: function(element_type) {
+        let display_index = 0;
+        if (!_.isEmpty(this.elements.models)) {
+            let lowest_element = _.max(this.elements.models, element => element.get('display_index'));
```
Comment (ConnorCimowsky) [inline src-backbone/app/js/models/page_model.js]: `lowest_element` seems kind of misleading here. lowest on the page, i.e., highest index? If so, maybe `highest_index_element` or `last_element`?

File: src-backbone/app/js/models/procedure_model.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            error: function() {
+                console.warn('Failed to create Page', page.get('id'));
+                // TODO show error alert
+            },
+        });
+    },
+
+    select_active_page: function(page_model) {
+        this.active_page_id = page_model.get('id');
+        this.trigger('change:active_page', page_model);
```
Comment (ConnorCimowsky) [inline src-backbone/app/js/models/procedure_model.js]: should this be pulled out into a constant like `REORDERED_ELEMENTS_EVENT`?

**Comment by ConnorCimowsky** [comment]
yes-thumbs-up-people

File: src-backbone/app/js/behaviors/sortable_collection_behavior.js (hunk 1 above)
Comment (Trinovantes) [inline src-backbone/app/js/behaviors/sortable_collection_behavior.js]: fixed

File: src-backbone/app/js/models/page_model.js (hunk 2 above)
Comment (Trinovantes) [inline src-backbone/app/js/models/page_model.js]: changed to `last_element`

File: src-backbone/app/js/models/procedure_model.js (hunk 3 above)
Comment (Trinovantes) [inline src-backbone/app/js/models/procedure_model.js]: this is now a const in the elements collection file

**Comment by Ommy** [comment]
Pulled your branch and launched it up and I see this inconsistent sizing for the buttons for elements:

image

File: src-backbone/app/js/views/procedures_builder/navbar.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    //-------------------------------------------------------------------------
+    // File menu
+    //-------------------------------------------------------------------------
+
+    events: {
+        'click #menu-bar ul.file a.save'  : 'file_save',
+        'click #menu-bar ul.file a.delete': 'file_delete',
+    },
+
+    file_save: function() {
```
Comment (Ommy) [inline src-backbone/app/js/views/procedures_builder/navbar.js]: Why `file_save` and `file_delete` as function names?

**Comment by Ommy** [comment]
Overall looks fine besides my comments about inconsistent buttons/boxes. I do have one question, why do we have a `File` menu. I don't think it's appropriate for the app. We can make it `Menu` or something else.

**Comment by Trinovantes** [comment]
The boxes have different heights because there's not enough space on the screen to evenly distribute the width. This should be fixed in another commit.

We have a File menu because in the future we may add an Edit, View, Help, etc. menus.

**Comment by Ommy** [comment]
Ah yeah that should dynamically change the boxes, but if that's being fixed that's good.

I was more referring to the name of it, but okay.

+1

**Pull request merged**